### ISSUE TITLE
Test the pre-trained model with EuRoc Vicon Room

### ISSUE THREAD

**Issue by Zchappie** [JiaRenChang/PSMNet#138]
HI,
Firstly, I want to appreciate your great work and share the code!
I try to use your pre-trained model KITTI 2015/2012 to test the `EuRoc Vicon Room` dataset, I have change the input channel to `1`, and padding to `(480, 768)`. And this is (one of them) what I got
![1403715329912143104](https://user-images.githubusercontent.com/29981524/60388285-7ededd80-9aaf-11e9-92a4-26ea0ab52b26.png)
Do you have any suggestions? Or the models are not suitable for other datasets before transfer learning?
Thanks

**Comment by JiaRenChang** [comment]
@Zchappie 
Hi.
Please notice your torchvision version, you should use 0.2.0.

**Comment by Zchappie** [comment]
Thanks for the fast reply!
I corrected the torchvision version, but the output remain the same.

**Comment by JiaRenChang** [comment]
@Zchappie 
Hi, would you provide me a pair of images to test?

**Comment by Zchappie** [comment]
Sure!
![1403715273262142976](https://user-images.githubusercontent.com/29981524/60413901-b72d0b80-9bd6-11e9-8eb5-68d394225b65.png)
![1403715273262142976](https://user-images.githubusercontent.com/29981524/60413914-cb710880-9bd6-11e9-8da7-0ffce1d7a1fe.png)

**Comment by JiaRenChang** [comment]
Hi, @Zchappie 
I found that this pair of images is not correctly rectified.
![test](https://user-images.githubusercontent.com/11732099/60525478-fff0db80-9d20-11e9-88e9-d5240fc419ba.png)

**Comment by Zchappie** [comment]
I did the rectification again, these are the new images.
![1403715273262142976](https://user-images.githubusercontent.com/29981524/60684045-c8f30500-9e9b-11e9-942d-84763e15aee8.png)
![1403715273262142976](https://user-images.githubusercontent.com/29981524/60684055-d14b4000-9e9b-11e9-8d4e-95e573f70be9.png)
But the output of disparity seems remain unchanged.

**Comment by JiaRenChang** [comment]
@Zchappie 
Here is my result of your image pair.
I used trained KITTI2015 model.

 
![test_color](https://user-images.githubusercontent.com/11732099/60888242-ebfd2a80-a288-11e9-9204-47159795b90a.png)

**Comment by Zchappie** [comment]
@JiaRenChang Cool! Could you please provide how you tune the params?

**Comment by JiaRenChang** [comment]
Hi, @Zchappie 
I didn't tune any parameters.
I used the KITTI 2015 pre-trained model. 
You can find the download link on Readme. 
The only difference is that I loaded the gray images as color images via opencv function.
You can modify the image load function to this:
yourimg = cv2.cvtColor(cv2.imread('imgpath',0), cv2.COLOR_GRAY2RGB) .

**Comment by Zchappie** [comment]
@JiaRenChang Thanks for the suggestions! 
Previously, I tested EuRoc dataset with single channel on my MacBook Pro, it worked (except the result is not correct). But since I loaded the images through OpenCV as you suggested, error `... not supported on CPUType for Byte` shows.  As discussed in https://github.com/JiaRenChang/PSMNet/issues/44#issuecomment-389499837, is that means it won't work for color images?

**Comment by JiaRenChang** [comment]
@Zchappie 
I did not really catch what you means.
My system is ubuntu 16.04, and an NVIDIA 1080 ti for inference.

**Comment by Zchappie** [comment]
@JiaRenChang Sorry, I mean do I need a GPU to do inference?

**Comment by JiaRenChang** [comment]
@Zchappie 
Yes, it would be better to have a GPU to infer.

**Comment by Zchappie** [comment]
@JiaRenChang Thanks for you patient explanations and suggestions!

**Comment by Zchappie** [comment]
@JiaRenChang Hi, I tried on Google Colab with CUDA Version 10.0.130, torchvision-0.2.0, torch-1.1.0, but I am not able to get the result like yours, at least not coloured disparity map. It is gray and similar to the very first post.
![1403715317412143104](https://user-images.githubusercontent.com/29981524/61542792-8bb57800-aa42-11e9-9d91-f51149acdfae.png)

**Comment by zhiwenfan** [comment]
@Zchappie  hi, I think you should follow the instruction, using pytorch0.4 and torchvision0.2.0

**Comment by Zchappie** [comment]
@zhiwenfan Thank you for the advise. I did, and it didn't work. That's why I try some other versions as well. I am totally new to set all those things up, but I have never expected it will be so complicated for me.

**Comment by JiaRenChang** [comment]
@Zchappie 
It may due to the version of skimage.
I am not sure.
My skimage version is 0.13.0.
scikit-image                       0.13.0

**Comment by Huyhust13** [comment]
I run submision.py use pretrained model for cityscapes dataset. Size of image (2048x1024) too large with my GPU (6GB). I resize to (1024x512) then it run ok but disparity map seem not good.
![image](https://user-images.githubusercontent.com/34271857/61773615-4d8fce00-ae1f-11e9-8d2c-e3d9d634344c.png)

**Comment by zhiwenfan** [comment]
> Hi, @Zchappie
> I found that this pair of images is not correctly rectified.
> ![test](https://user-images.githubusercontent.com/11732099/60525478-fff0db80-9d20-11e9-88e9-d5240fc419ba.png)

hi @JiaRenChang , could you please provide me your python code to draw this image? many thanks!

**Comment by mike132129** [comment]
> I did the rectification again, these are the new images.
> ![1403715273262142976](https://user-images.githubusercontent.com/29981524/60684045-c8f30500-9e9b-11e9-942d-84763e15aee8.png)
> ![1403715273262142976](https://user-images.githubusercontent.com/29981524/60684055-d14b4000-9e9b-11e9-8d4e-95e573f70be9.png)
> But the output of disparity seems remain unchanged.

Hi @Zchappie, did you have a better result?
I am trying to test the model with my grayscale image, but the result seems to be not good.
And I test your image pair, getting a whole blue disparity map, not like the author's.

**Comment by huangju91** [comment]
Hi,I just scanned your chat log, I wonder if the stereo rectification for the test image pair is necessary?

**Comment by ynma-hanvo** [comment]
nice chat thread;    I am wondering how long does it take for 768 * 480 depth inference , and 
is the output depth value   the pixel shift between the left and right image ?
